src/iarc7_motion/iarc_tasks/takeoff_task.py

```diff
@@ -106,39 +87,21 @@ def get_desired_command(self):
         if self._state == TakeoffTaskState.pause_before_takeoff:
             if rospy.Time.now() - self.pause_start_time > rospy.Duration(self._DELAY_BEFORE_TAKEOFF):
                 self._state = TakeoffTaskState.takeoff
+                return (TaskRunning(), GroundInteractionCommand(
+                                       'takeoff',
+                                       self.takeoff_callback))
             else:
-                return (TaskRunning(), VelocityCommand())
+                return (TaskRunning(), NopCommand())
 
         # Enter the takeoff phase
         if self._state == TakeoffTaskState.takeoff:
-            try:
-                transStamped = self._tf_buffer.lookup_transform(
-                                    'map',
-                                    'quad',
-                                    rospy.Time.now(),
-                                    rospy.Duration(self._TRANSFORM_TIMEOUT))
-            except (tf2_ros.LookupException,
-                    tf2_ros.ConnectivityException,
-                    tf2_ros.ExtrapolationException) as ex:
-                msg = 'Exception when looking up transform during takeoff'
-                rospy.logerr('Takeofftask: {}'.format(msg))
-                rospy.logerr(ex.message)
-                return (TaskAborted(msg=msg),)
-
-            # Check if we reached the target height
-            if(transStamped.transform.translation.z > self._takeoff_height - self._TAKEOFF_HEIGHT_TOLERANCE):
-                return (TaskDone(), VelocityCommand())
-
-            # Check if we are above maneuver height
-            if(transStamped.transform.translation.z > self._MIN_MANEUVER_HEIGHT):
-                hold_twist = self._path_holder.get_xy_hold_response(z_velocity=self._TAKEOFF_VELOCITY)
-                return (TaskRunning(), VelocityCommand(hold_twist))
-            else:
-                velocity = TwistStamped()
-                velocity.header.frame_id = 'level_quad'
-                velocity.header.stamp = rospy.Time.now()
-                velocity.twist.linear.z = self._TAKEOFF_VELOCITY
-                return (TaskRunning(), VelocityCommand(velocity))
+            return (TaskRunning(),)
```

Comment:
No you aren't required to return a command if you don't want to. A nop command is a zero velocity command and I don't believe I actually want to send that here.

NopCommand's should probably have a different name.